allensdk/brain_observatory/behavior/behavior_session.py

```diff
@@ -9,53 +10,121 @@
 from allensdk.brain_observatory.behavior.session_apis.abcs.\
     session_base.behavior_base import BehaviorBase
 from allensdk.brain_observatory.behavior.trials_processing import (
     construct_rolling_performance_df, calculate_reward_rate_fix_nans)
+from allensdk.brain_observatory.behavior.data_objects import (
+    BehaviorSessionId, StimulusTimestamps, RunningSpeed, RunningAcquisition
+)
+
+from allensdk.core.auth_config import LIMS_DB_CREDENTIAL_MAP
+from allensdk.internal.api import db_connection_creator
 
 
 BehaviorDataApi = Type[BehaviorBase]
 
 
 class BehaviorSession(LazyPropertyMixin):
-    def __init__(self, api: Optional[BehaviorDataApi] = None):
+    def __init__(
+        self, api: Optional[BehaviorDataApi] = None,
+        behavior_session_id: BehaviorSessionId = None,
+        stimulus_timestamps: StimulusTimestamps = None,
+        running_acquisition: RunningAcquisition = None,
+        raw_running_speed: RunningSpeed = None,
+        running_speed: RunningSpeed = None,
+    ):
         self.api = api
 
         # LazyProperty constructor provided by LazyPropertyMixin
         LazyProperty = self.LazyProperty
 
         # Initialize attributes to be lazily evaluated
-        self._behavior_session_id = LazyProperty(
-            self.api.get_behavior_session_id)
+        self._behavior_session_id = behavior_session_id
         self._licks = LazyProperty(self.api.get_licks, settable=True)
         self._rewards = LazyProperty(self.api.get_rewards, settable=True)
-        self._running_speed = LazyProperty(self.api.get_running_speed,
-                                           settable=True, lowpass=True)
-        self._raw_running_speed = LazyProperty(self.api.get_running_speed,
-                                               settable=True, lowpass=False)
+        self._running_acquisition = running_acquisition
+        self._running_speed = running_speed
+        self._raw_running_speed = raw_running_speed
         self._stimulus_presentations = LazyProperty(
             self.api.get_stimulus_presentations, settable=True)
         self._stimulus_templates = LazyProperty(
-            self.api.get_stimulus_templates, settable=True)
-        self._stimulus_timestamps = LazyProperty(
-            self.api.get_stimulus_timestamps, settable=True)
+            self.api.get_stimulus_templates, settable=True
+        )
+        self._stimulus_timestamps = stimulus_timestamps
         self._task_parameters = LazyProperty(self.api.get_task_parameters,
                                              settable=True)
         self._trials = LazyProperty(self.api.get_trials, settable=True)
         self._metadata = LazyProperty(self.api.get_metadata, settable=True)
 
     # ==================== class and utility methods ======================
 
+    @classmethod
+    def from_json(cls, session_data: dict) -> "BehaviorSession":
+        behavior_session_id = BehaviorSessionId.from_json(session_data)
```

Comment:
technically, these methods are `from_dict()` (and `to_dict()`).
to me `json` implies a file, not an object.

Reply:
I think `from_json()` and `to_json()` also suggests that the `dict` cannot have arbitrary python objects that cannot be serialized to JSON

Reply:
good point

Reply:
`to_json_compatible_dict` is a little wordy, but, that is the implication